src/libraries/Common/src/System/Security/Cryptography/SlhDsaImplementation.cs

```diff
@@ -11,9 +11,29 @@ internal sealed partial class SlhDsaImplementation : SlhDsa
     {
         internal static partial bool SupportsAny();
 
-        internal static partial SlhDsa GenerateKeyCore(SlhDsaAlgorithm algorithm);
-        internal static partial SlhDsa ImportPublicKey(SlhDsaAlgorithm algorithm, ReadOnlySpan<byte> source);
-        internal static partial SlhDsa ImportPkcs8PrivateKeyValue(SlhDsaAlgorithm algorithm, ReadOnlySpan<byte> source);
-        internal static partial SlhDsa ImportSecretKey(SlhDsaAlgorithm algorithm, ReadOnlySpan<byte> source);
+        internal static partial SlhDsaImplementation GenerateKeyCore(SlhDsaAlgorithm algorithm);
+        internal static partial SlhDsaImplementation ImportPublicKey(SlhDsaAlgorithm algorithm, ReadOnlySpan<byte> source);
+        internal static partial SlhDsaImplementation ImportPkcs8PrivateKeyValue(SlhDsaAlgorithm algorithm, ReadOnlySpan<byte> source);
+        internal static partial SlhDsaImplementation ImportSecretKey(SlhDsaAlgorithm algorithm, ReadOnlySpan<byte> source);
+
+        /// <summary>
+        ///   Duplicates an SLH-DSA private key by export/import.
+        ///   Only intended to be used when the key type is unknown.
+        /// </summary>
+        internal static SlhDsaImplementation DuplicatePrivateKey(SlhDsa key)
+        {
+            Debug.Assert(key is not SlhDsaImplementation);
+
+            // TODO Does this need to use CryptoPool?
```

Comment:
It can, and thus probably should.

Reply:
Should just change this to a 128 byte stackalloc and pass in the sliced span.